source/event_hub.ts

```diff
@@ -606,20 +606,24 @@ export class EventHub {
    * Publish reply event.
    * @param  {Object} sourceEventPayload Source event payload
    * @param  {Object} data        Response event data
-   * @param  {Object} [source]    Response event source information
+   * @param  {Object} source      Deprecated value - will be removed in a later version
    */
   publishReply(
     sourceEventPayload: EventPayload,
     data: Data,
     source: Data | null = null
   ): Promise<string> {
-    const replyEvent = new Event("ftrack.meta.reply", {
-      ...data,
-      target: `id=${sourceEventPayload.source.id}`,
-      inReplyToEvent: sourceEventPayload.id,
-      source: source ?? data.source,
-    });
-
+    const replyEvent = new Event(
+      "ftrack.meta.reply",
+      {
+        ...data,
+      },
+      {
+        target: `id=${sourceEventPayload.source.id}`,
+        inReplyToEvent: sourceEventPayload.id,
+        source: source ?? data.source,
```

Comment:
I don't think we should to default to `data.source`, since source should be defined as Event.source, not Event.data.source.

See the [previous JS implementation](https://github.com/ftrackhq/ftrack-javascript/blob/0.9.0/source/event_hub.js#L408-L417) for reference.

Reply:
You are correct, didn't compare this part to the old implementation, just copied what we have. Good find.